### PR TITLE
Fix the host-specific installation destdir for build-script products.

### PR DESCRIPTION
**Pull request by DougGregor**
Prior to this, build-script products that build in the host-specific
directories for cross-compilation would get the wrong path, and
therefore wouldn't get picked up in toolchain builds.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please build toolchain

File: utils/swift_build_support/swift_build_support/products/product.py [hunk 1]
```diff
@@ -224,7 +224,7 @@ def host_install_destdir(self, host_target):
                 # If this is one of the hosts we should lipo,
                 # install in to a temporary subdirectory.
                 return '%s/intermediate-install/%s' % \
-                    (self.args.install_destdir, host_target)
+                    (os.path.dirname(self.build_dir), host_target)
```
Comment (buttaface) [inline utils/swift_build_support/swift_build_support/products/product.py:227]: Heh, I had a fix for this in another pull, #36917.

Comment (DougGregor) [reply utils/swift_build_support/swift_build_support/products/product.py:227]: Hah, I wish I'd seen that before! It's totally the fix for my problem, too

**Pull request merged**